### PR TITLE
[dev-2.5] Change CSI tolerations to match upstream and allow override

### PR DESCRIPTION
**Pull request by PennyScissors**
Issue: https://github.com/rancher/rancher/issues/34751

This PR is basically the same as [this PR](https://github.com/rancher/charts/pull/1473), but it was reverted in 2.6.1 so we are adding it in 2.5.10 and forward porting it instead. It also adds a kubeVersion to match their compatibility matrix [here](https://vsphere-csi-driver.sigs.k8s.io/compatiblity_matrix.html#compatibility-matrix-for-vsphere-csi-driver).

### CODE REVIEW

**Comment by PennyScissors** [comment]
This PR is the same thing as [this](https://github.com/rancher/charts/pull/1473), but it was reverted in 2.6.1 so we are adding it in 2.5.10 and forward porting it instead.

File: packages/rancher-vsphere-csi/charts/templates/controller/deployment.yaml [hunk 1]
```diff
@@ -39,25 +39,17 @@ spec:
...
+      {{- with .Values.csiController.tolerations }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- else }}
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
           effect: NoSchedule
-        - operator: "Exists"
-          effect: NoExecute
```
Comment (jiaqiluo) [inline packages/rancher-vsphere-csi/charts/templates/controller/deployment.yaml:50]: Why is this one dropped?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: packages/rancher-vsphere-csi/charts/templates/controller/deployment.yaml (hunk 1 above)
Comment (PennyScissors) [reply packages/rancher-vsphere-csi/charts/templates/controller/deployment.yaml:50]: To match what's [upstream](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/e878840857838f84e89784823cdbbd5d6b1c853a/manifests/vanilla/vsphere-csi-driver.yaml#L205-L218)

Comment (jiaqiluo) [reply packages/rancher-vsphere-csi/charts/templates/controller/deployment.yaml:50]: It makes sense to me now.

**Review by jiaqiluo — APPROVED** [review]

File: packages/rancher-vsphere-csi/charts/values.yaml [hunk 2]
```diff
@@ -70,6 +81,7 @@ csiNode:
     livenessProbe:
       repository: rancher/mirrored-k8scsi-livenessprobe
       tag: v2.2.0
+  tolerations: {}
```
Comment (aiyengar2) [inline packages/rancher-vsphere-csi/charts/values.yaml:84]: why did we introduce tolerations here?

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

File: packages/rancher-vsphere-csi/charts/values.yaml (hunk 2 above)
Comment (PennyScissors) [reply packages/rancher-vsphere-csi/charts/values.yaml:84]: I left that in there by mistake, I'll remove it.

**Review by aiyengar2 — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**